### ISSUE TITLE
Local variable 'body' can't be referenced before it is declared.

### ISSUE THREAD

**Issue by edwardaux** [lejard-h/chopper#12]
If I declare a service that looks like:

```
@ChopperApi()
abstract class MyService extends ChopperService {
  static MyService create([ChopperClient client]) => _$MyService(client);

  @Post(url: 'my/url')
  @FormUrlEncoded()
  Future<Response<MyInfo>> fetchSomething(@Body() Map<String, dynamic> body);
}
```

I get an error saying `Local variable 'body' can't be referenced before it is declared.` 

The underlying reason appears to be that the generated code has a local variable called `body` and we end up with generated code that looks like:

```
final body = body;
...
```

I can easily work around it by renaming the `fetchSomething` parameter to be called something other than `body`.

**Comment by lejard-h** [comment]
Indeed I need to rename variables in the generated code `body` => `$body` to avoid collision

**Comment by lejard-h** [comment]
fix in `1.2.0-dev3`
